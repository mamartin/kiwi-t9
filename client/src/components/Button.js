// @flow
import React from "react"
import { TouchableOpacity, Text } from "react-native"
import Colors from "../themes/Colors"

const styles = {
  button: {
    alignItems: "center",
    backgroundColor: Colors.primary,
    borderRadius: 15,
    height: 60,
    justifyContent: "center",
    width: "30%",
    margin: 3,
  },
  text: {
    color: "white",
  },
  number: {
    fontWeight: "bold",
    fontSize: 20,
    color: "white",
  },
}

type ButtonProps = {|
  onPress: () => void,
  children: string,
  number: string,
|}

export default class Button extends React.PureComponent<ButtonProps> {
  render() {
    const { children, onPress, number } = this.props
    return (
      <TouchableOpacity onPress={onPress} style={styles.button}>
        <Text style={styles.number}>{number}</Text>
        <Text style={styles.text}>{children}</Text>
      </TouchableOpacity>
    )
  }
}
