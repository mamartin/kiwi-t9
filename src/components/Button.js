// @flow
import React from "react"
import { TouchableOpacity, Text } from "react-native"

const styles = {
  button: {
    alignItems: "center",
    backgroundColor: "grey",
    borderRadius: 60,
    height: 60,
    justifyContent: "center",
  },
  text: {
    color: "white",
  },
}

type ButtonProps = {|
  onPress: () => void,
  children: string | React.DOM,
|}

export default class Button extends React.PureComponent<ButtonProps> {
  render() {
    const { children, onPress } = this.props
    return (
      <TouchableOpacity onPress={onPress} style={styles.button}>
        {typeof children === "string" ? (
          <Text style={styles.text}>{children}</Text>
        ) : (
          children
        )}
      </TouchableOpacity>
    )
  }
}
