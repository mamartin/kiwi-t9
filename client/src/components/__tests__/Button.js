import React from "react"
import renderer from "react-test-renderer"

// components
import Button from "../Button"

describe("Button component", () => {
  it("renders with default props", () => {
    const tree = renderer.create(<Button>Button</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
