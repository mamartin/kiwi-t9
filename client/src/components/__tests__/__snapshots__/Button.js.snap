// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders with default props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(115,115,115)",
      "borderRadius": 15,
      "height": 60,
      "justifyContent": "center",
      "margin": 3,
      "opacity": 1,
      "width": "30%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  >
    Button
  </Text>
</View>
`;
